src/hex_view/mod.rs

```diff
@@ -0,0 +1,274 @@
+use crossterm::style::Attributes;
+use crossterm::{
+    queue,
+    style::{self, Color},
+    ErrorKind,
+};
+use std::fmt;
+use std::fmt::Display;
+use std::io::Write;
+
+mod byte_properties;
+pub mod view;
+
+const COLOR_NULL: Color = Color::AnsiValue(150);
+const COLOR_ASCII_PRINTABLE: Color = Color::Cyan;
+const COLOR_ASCII_WHITESPACE: Color = Color::Green;
+const COLOR_ASCII_OTHER: Color = Color::Rgb {
+    r: 232,
+    g: 52,
+    b: 210,
+};
+const COLOR_NONASCII: Color = Color::Yellow;
+
+#[derive(Debug, Clone, Copy)]
+pub enum Priority {
+    Basic,
+    #[allow(dead_code)]
+    Mark,
+    Selection,
+    Cursor,
+}
+
+#[derive(Debug, Clone)]
+pub struct PrioritizedStyle {
+    style: style::ContentStyle,
+    #[allow(dead_code)]
+    priority: Priority,
+}
+
+#[derive(Debug, Clone)]
+pub struct StylingCommand {
+    start: Option<PrioritizedStyle>,
+    mid: Option<PrioritizedStyle>,
+    end: Option<PrioritizedStyle>,
+}
+
+impl Default for StylingCommand {
+    fn default() -> Self {
+        Self {
+            start: Some(PrioritizedStyle {
+                style: style::ContentStyle {
+                    foreground_color: Some(Color::White),
+                    background_color: Some(Color::Reset),
+                    attributes: Attributes::default(),
+                },
+                priority: Priority::Basic,
+            }),
+            mid: None,
+            end: Some(PrioritizedStyle {
+                style: style::ContentStyle {
+                    foreground_color: Some(Color::White),
+                    background_color: Some(Color::Reset),
+                    attributes: Attributes::default(),
+                },
+                priority: Priority::Basic,
+            }),
+        }
+    }
+}
```

Comment:
This must be the origin of the bug I've mentioned. The "Default" styling command should in fact be a no-op, rather than specifying a change to a different style. (Does that make sense? The language with "Default" is really confusing, as it would seem that this here is the default style for uncolored bytes.)

Anyway, going back to the `#derive`d `Default` implementation seemed to fix the highlighting.

Reply:
Yes, it was a cause.
The purpose of the custom Default implementation was to fix the styling of the separator on the last line. I revert it to `#derive` and fixed styling in another way. Also, I made the selection background darker (to improve readability). See my last commit